src/element/mod.rs

```diff
@@ -348,6 +349,50 @@ impl std::fmt::Debug for Element {
     }
 }
 
+impl TryFrom<Element> for Sequence {
+    type Error = IonError;
+
+    fn try_from(element: Element) -> Result<Sequence, Self::Error> {
+        match element.try_into_sequence() {
+            ControlFlow::Continue(sequence) => Ok(sequence),
+            ControlFlow::Break(element) => Err(element.expected("sequence value")),
+        }
+    }
+}
```

Comment:
You could do something like this, to DRY some of the boilerplate and also implement `TryFrom<Element>` for more types so that all Ion types are covered. (I think it's a good idea to add it for all of them so that we're not asking later why we didn't do that. The codegen project, in particular, really benefits from having consistent APIs that cover all the types.)
```rust
macro_rules! impl_try_from_element {
    ($to_type:ident, $to_fn:ident, $to_desc:expr) => {
        impl TryFrom<Element> for $to_type {
            type Error = IonError;

            fn try_from(element: Element) -> Result<$to_type, Self::Error> {
                match element.$to_fn() {
                    ControlFlow::Continue(structure) => Ok(structure),
                    ControlFlow::Break(element) => Err(element.expected($to_desc)),
                }
            }
        }
    }
}

impl_try_from_element!(Struct, try_into_struct, "struct value")
impl_try_from_element!(Sequence, try_into_sequence, "sequence value")
impl_try_from_element!(String, try_into_text, "text value")
impl_try_from_element!(Str, try_into_string, "string value")
impl_try_from_element!(Symbol, try_into_symbol, "symbol value")
impl_try_from_element!(bool, try_into_bool, "boolean value")
impl_try_from_element!(f64, try_into_float, "float value")
impl_try_from_element!(Decimal, try_into_decimal, "decimal value")
// etc.
```